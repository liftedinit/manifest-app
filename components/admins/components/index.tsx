export * from './validatorList';
export * from './stakeHolderPayout';
export * from './chainUpgrader';
