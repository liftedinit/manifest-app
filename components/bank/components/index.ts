export * from './sendBox';
export * from './tokenList';
export * from './historyBox';
