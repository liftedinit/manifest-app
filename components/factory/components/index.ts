export * from './DenomList';
export * from './DenomImage';
