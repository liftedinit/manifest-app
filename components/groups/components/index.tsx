export * from './CountdownTimer';
export * from '../modals/groupInfo';
export * from './groupProposals';
export * from './myGroups';
export * from '../../react/StepIndicator';
