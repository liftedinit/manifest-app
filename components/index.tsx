export * from './types';
export * from './react';
export * from './groups';
export * from './wallet';
export * from './admins';
export * from './factory';
export * from './bank';
export * from './icons';
export * from './divider';
export * from './seo';
export * from './messageSyntax';
