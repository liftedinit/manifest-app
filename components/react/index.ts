export * from './addressCopy';
export * from './modal';
export * from './views';
export * from './settingsModal';
export * from './inputs';
export { SignModal } from './authSignerModal';
export * from './ModalDialog';
