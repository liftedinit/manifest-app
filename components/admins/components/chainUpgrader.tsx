import { UpgradeModal, CancelUpgradeModal } from '@/components/admins/modals';
import { useCurrentPlan } from '@/hooks/useQueries';
import { PlanSDKType } from '@liftedinit/manifestjs/dist/codegen/cosmos/upgrade/v1beta1/upgrade';
import { useState } from 'react';

export const ChainUpgrader = ({ admin, address }: { admin: string; address: string }) => {
  const [isOpen, setIsOpen] = useState(false);
  const [isCancelOpen, setIsCancelOpen] = useState(false);
  const { plan, isPlanLoading, refetchPlan } = useCurrentPlan();

  if (isPlanLoading) {
    return (
      <div className="w-full md:w-1/2 h-full bg-secondary relative rounded-lg p-6 flex flex-col gap-4 shadow-lg animate-pulse">
        <div className="absolute top-4 right-4">
          <div className="skeleton h-4 w-32"></div>
        </div>
        <div className="space-y-2">
          <div className="skeleton h-7 w-64"></div>
          <div className="skeleton h-4 w-full"></div>
        </div>
        <div className="flex flex-row w-full justify-between gap-4 mt-4">
          <div className="skeleton h-10 flex-1"></div>
          <div className="skeleton h-10 flex-1"></div>
        </div>
      </div>
    );
  }

  return (
    <div className="w-full md:w-1/2 h-full bg-secondary relative rounded-lg p-6 flex flex-col gap-4 shadow-lg">
      <div className="absolute top-4 right-4">
        <p className={`text-xs ${plan ? 'text-success' : 'text-content'}`}>
          {plan ? 'Upgrade in progress' : 'No upgrade in progress'}
        </p>
      </div>
      <div className="space-y-2">
        <h1 className="text-xl font-bold text-secondary-content">Submit Chain Upgrade Proposal</h1>
        <p className="text-secondary-content/80 text-sm">
          Submit a proposal to instantiate a chain upgrade.
        </p>
      </div>

      <div className="flex flex-row w-full justify-between gap-4 mt-4">
        <button
          disabled={!!plan}
          onClick={() => setIsOpen(true)}
          className="btn btn-primary flex-1"
        >
          Upgrade Chain
        </button>
        <button
          disabled={!plan}
          onClick={() => setIsCancelOpen(true)}
          className="btn btn-error flex-1 dark:text-white text-black"
        >
          Cancel Upgrade
        </button>
      </div>
      <UpgradeModal
        address={address}
        admin={admin}
        isOpen={isOpen}
        onClose={() => setIsOpen(false)}
        refetchPlan={refetchPlan}
      />
      <CancelUpgradeModal
        plan={plan ?? ({} as PlanSDKType)}
        admin={admin}
        address={address}
        isOpen={isCancelOpen}
        onClose={() => setIsCancelOpen(false)}
        refetchPlan={refetchPlan}
      />
    </div>
  );
};
