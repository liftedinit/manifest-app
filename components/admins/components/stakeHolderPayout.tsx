import { MultiMintModal, MultiBurnModal } from '@/components/factory/modals';
import { MFX_TOKEN_DATA } from '@/utils/constants';
import { useState } from 'react';

interface StakeHolderPayoutProps {
  admin: string;
  address: string;
}

export const StakeHolderPayout = ({ admin, address }: StakeHolderPayoutProps) => {
  const [isOpenMint, setIsOpenMint] = useState(false);
  const [isOpenBurn, setIsOpenBurn] = useState(false);

  return (
    <div className="w-full md:w-1/2 h-full bg-secondary rounded-lg p-6 flex flex-col gap-4 shadow-lg">
      <div className="space-y-2">
        <h1 className="text-xl font-bold text-secondary-content">Stake Holder Payout</h1>
        <p className="text-secondary-content/80 text-sm">
          Burn MFX or mint MFX to pay out stake holders.
        </p>
      </div>

      <div className="flex flex-row w-full justify-between gap-4 mt-4">
        <button onClick={() => setIsOpenMint(true)} className="btn btn-primary flex-1">
          Mint MFX
        </button>
        <button
          onClick={() => setIsOpenBurn(true)}
          className="btn btn-error flex-1 dark:text-white text-black"
        >
          Burn MFX
        </button>
      </div>
      <MultiMintModal
        isOpen={isOpenMint}
        onClose={() => setIsOpenMint(false)}
        admin={admin}
        address={address}
        denom={MFX_TOKEN_DATA}
      />
      <MultiBurnModal
        isOpen={isOpenBurn}
        onClose={() => setIsOpenBurn(false)}
        admin={admin}
        address={address}
        denom={MFX_TOKEN_DATA}
      />
    </div>
  );
};
