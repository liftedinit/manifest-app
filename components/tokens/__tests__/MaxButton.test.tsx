import { cleanup, render, screen } from '@testing-library/react';
import { afterEach, describe, expect, jest, mock, test } from 'bun:test';
import React from 'react';

import { MaxButton } from '@/components';
import { formatComponent } from '@/tests';
import { mockBalances, mockMfxBalance } from '@/tests/data';

describe('MaxButton', () => {
  afterEach(cleanup);

  test('works', () => {
    const token = mockBalances[0];

    let amount = 0;
    const wrapper = render(
      <MaxButton token={token} setTokenAmount={a => (amount = parseFloat(a))} />
    );

    expect(amount).toBe(0);
    expect(screen.getByText('MAX')).toBeInTheDocument();
    screen.getByText('MAX').click();
    expect(amount).toBe(0.001);
    expect(document.querySelector('.tooltip')).not.toBeInTheDocument();

    expect(formatComponent(wrapper.asFragment())).toMatchSnapshot();
  });

  test('works for MFX', () => {
    const token = mockMfxBalance;

    let amount = 0;
    const wrapper = render(
      <MaxButton token={token} setTokenAmount={a => (amount = parseFloat(a))} />
    );

    expect(amount).toBe(0);
    expect(screen.getByText('MAX')).toBeInTheDocument();
    screen.getByText('MAX').click();
    expect(amount).toBe(1.9);
    expect(document.querySelector('.tooltip')).toBeInTheDocument();
  });
});
