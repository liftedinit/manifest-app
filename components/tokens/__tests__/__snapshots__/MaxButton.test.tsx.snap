// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`MaxButton works 1`] = `
"<DocumentFragment>
  <button
    class="text-xs text-primary cursor-pointer"
    type="button"
  >
    MAX
  </button>
</DocumentFragment>"
`;
